### PR TITLE
updating docker volume resource to relfect new fields and structure

### PR DESCRIPTION
**Pull request by yunhee-l**
### Summary
Updating volume resource definition

### Implementation details
- renamed the volume.go to dockervolume.go 
- restructured to pull docker volume specific fields to dockerVolumeConfig struct
- added extra fields
- considered refactoring it now to make the VolumeConfig generic and not fixed to DockerVolumeConfig but decided to push that to later since we don't know if we will reuse this volume definition or not for V2.

### Testing

- [x] Builds on Linux (`make release`)
- [x] Builds on Windows (`go build -out amazon-ecs-agent.exe ./agent`)
- [x] Unit tests on Linux (`make test`) pass
- [ ] Unit tests on Windows (`go test -timeout=25s ./agent/...`) pass
- [x] Integration tests on Linux (`make run-integ-tests`) pass
- [x] Integration tests on Windows (`.\scripts\run-integ-tests.ps1`) pass
- [x] Functional tests on Linux (`make run-functional-tests`) pass
- [x] Functional tests on Windows (`.\scripts\run-functional-tests.ps1`) pass

New tests cover the changes: yes

### Licensing

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: agent/taskresource/volume/dockervolume.go [hunk 1]
```diff
@@ -38,19 +34,36 @@ type VolumeResource struct {
 	lock sync.RWMutex
 }
 
+// DockerVolumeConfig represents docker volume configuration
+type DockerVolumeConfig struct {
+	Scope          string `json:"scope"`
```
Comment (petderek) [inline agent/taskresource/volume/dockervolume.go]: `go fmt` plz

File: agent/taskresource/volume/dockervolume.go [hunk 2]
```diff
@@ -38,19 +34,36 @@ type VolumeResource struct {
 	lock sync.RWMutex
 }
 
+// DockerVolumeConfig represents docker volume configuration
```
Comment (petderek) [inline agent/taskresource/volume/dockervolume.go]: Consider:

a) linking to docs explaining what these are
b) commenting on each field

File: agent/taskresource/volume/dockervolume.go [hunk 3]
```diff
@@ -38,19 +34,36 @@ type VolumeResource struct {
...
 	client dockerapi.DockerClient) *VolumeResource {
 
 	return &VolumeResource{
-		Name:       name,
-		Driver:     driver,
-		DriverOpts: driverOptions,
-		Labels:     labels,
-		client:     client,
+		Name: name,
+		VolumeConfig: DockerVolumeConfig{
```
Comment (petderek) [inline agent/taskresource/volume/dockervolume.go]: This is awesome!

File: agent/taskresource/volume/dockervolume.go [hunk 4]
```diff
@@ -22,14 +22,10 @@ import (
...
-// VolumeResource represents docker volume resource
+// VolumeResource represents volume resource
 type VolumeResource struct {
-	Name string
-	// mountpoint is a read-only field returned from docker
-	mountpoint          string
-	Driver              string
-	DriverOpts          map[string]string
-	Labels              map[string]string
+	Name                string
```
Comment (aaithal) [inline agent/taskresource/volume/dockervolume.go]: minor: lint will tell you that all of the exported fields need to be documented

File: agent/taskresource/volume/dockervolume.go (hunk 1 above)
Comment (yunhee-l) [reply agent/taskresource/volume/dockervolume.go]: go fmt doesn't actually change this, interspersed comment line might be why? it doesn't align the field's name type json-tag columns.

File: agent/taskresource/volume/dockervolume.go (hunk 2 above)
Comment (yunhee-l) [reply agent/taskresource/volume/dockervolume.go]: Added link to the docker API for volume, added comments for 2 fields we introduced (scope and autoprovision).

**Pull request merged**